### ISSUE TITLE
gitfast: getting color in RPROMPT with git-prompt.sh

### ISSUE THREAD

**Issue by fniessen** [robbyrussell/oh-my-zsh#4088]
Though I'm setting `GIT_PS1_SHOWCOLORHINTS` to `true`, I don't get any color in the RPROMPT when sourcing `~/.oh-my-zsh/plugins/gitfast/git-prompt.sh`.

See http://screencast.com/t/dm0XvSDHE:
![image](https://user-images.githubusercontent.com/1441704/55590876-71c28480-5734-11e9-957d-1b9c1ce0d95e.png)

PS- Why am I using Git-prompt while I do already have much in the "left" prompt??  Because there are missing info there, such as the fact we're ahead of `origin/master` and the number of commits we will push next time we type `git push`. Of course, it's overkill to repeat the branch name and so on, but I intend to comment those in `git-prompt.sh`.

**Comment by apjanke** [comment]
The `__git_ps1()` function seems to [only do colorization when it's in "pcmode"](https://github.com/robbyrussell/oh-my-zsh/blob/599831b85a5fd4564ab820530e2bb45f37ca3f99/plugins/gitfast/git-prompt.sh#L443-L446), where it's expecting to set the PS1 string, instead of being used inside command substitution.

```
	# NO color option unless in PROMPT_COMMAND mode
	if [ $pcmode = yes ] && [ -n "${GIT_PS1_SHOWCOLORHINTS-}" ]; then
		__git_ps1_colorize_gitstring
	fi
```

To incorporate it in your custom `$PS1` or `$RPROMPT` with colorization intact, you'll need the function signature to change to allow enabling colorization independent of `pcmode`.

@felipec, you're listed as the `gitfast` plugin maintainer. Comments?

**Comment by fniessen** [comment]
Are there any reasons for limiting the colorization just for the PS1 prompt?

**Comment by apjanke** [comment]
Not that I know of. There are plenty of themes that use color in the `$RPROMPT`. It seems like `__git_ps`()` simply didn't foresee getting called this way.

**Comment by fniessen** [comment]
Following your implicit advice, I removed the test about `pcmode`, and that works perfectly.  Thanks!

Here is the patch:

    diff --git a/plugins/gitfast/git-prompt.sh b/plugins/gitfast/git-prompt.sh
    index 5448908..a89a7bc 100644
    --- a/plugins/gitfast/git-prompt.sh
    +++ b/plugins/gitfast/git-prompt.sh
    @@ -1,6 +1,6 @@
     # bash/zsh git prompt support
     #
    -# Copyright (C) 2006,2007 Shawn O. Pearce <spearce@spearce.org>
    +# Copyright (C) 2006,2007,2015 Shawn O. Pearce <spearce@spearce.org>
     # Distributed under the GNU General Public License, version 2.0.
     #
     # This script allows you to see repository status in your prompt.
    @@ -441,7 +441,7 @@ __git_ps1 ()
            local z="${GIT_PS1_STATESEPARATOR-" "}"

            # NO color option unless in PROMPT_COMMAND mode
    -	if [ $pcmode = yes ] && [ -n "${GIT_PS1_SHOWCOLORHINTS-}" ]; then
    +	if [ -n "${GIT_PS1_SHOWCOLORHINTS-}" ]; then
                    __git_ps1_colorize_gitstring
            fi

**Comment by felipec** [comment]
I'm not entirely sure, but it's probably a limitation of Bash. Maybe for zsh the check can be disabled?

**Comment by felipec** [comment]
This has actually been fixed since a long while (079fef48).

**Comment by fniessen** [comment]
Thanks for the note!
